fabric-biomes-v1/src/main/java/net/fabricmc/fabric/impl/biome/InternalBiomeData.java

```diff
@@ -134,6 +141,24 @@ public static void addNetherBiome(Biome biome, Biome.MixedNoisePoint spawnNoiseP
 		NETHER_BIOME_NOISE_POINTS.put(biome, spawnNoisePoint);
 	}
 
+	public static void addFeatureToBiome(Biome biome, GenerationStep.Feature generationStep, ConfiguredFeature<?, ?> configuredFeature) {
+		List<List<Supplier<ConfiguredFeature<?, ?>>>> features = biome.method_30970().method_30983();
+
+		int generationStepIndex = generationStep.ordinal();
+
+		while (features.size() <= generationStepIndex) {
+			features.add(Lists.newArrayList());
+		}
+
+		List<Supplier<ConfiguredFeature<?, ?>>> stepList = features.get(generationStepIndex);
+
+		if (stepList instanceof ImmutableList) {
+			features.set(generationStepIndex, stepList = new ArrayList<>(stepList));
```

Comment:
Hmm, do we have rule on whether put assignments in individual statements? Though this type of call is indeed better performance wise

Reply:
I did this to save 2 lines of code (new vartible, add to map, reassign exisitng var) its not a super clear way of doing thing, but I think its fine here